bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -95,14 +118,57 @@ public Response httpGetPersistenceServices(@Context HttpHeaders headers) {
     public Response httpGetPersistenceItemData(@Context HttpHeaders headers,
-            @ApiParam(value = "The item name", required = true) @PathParam("itemname") String itemName,
             @ApiParam(value = "Name of the persistence service. If not provided the default service will be used", required = false) @QueryParam("servicename") String serviceName,
+            @ApiParam(value = "The item name", required = true) @PathParam("itemname") String itemName,
             @ApiParam(value = "Start time of the data to return. Will default to 1 day before endtime", required = false) @QueryParam("starttime") String startTime,
             @ApiParam(value = "End time of the data to return. Will default to current time.", required = false) @QueryParam("endtime") String endTime,
             @ApiParam(value = "Page number of data to return. This parameter will enable paging.", required = false) @QueryParam("page") int pageNumber,
-            @ApiParam(value = "The length of each page.", required = false) @QueryParam("pagelength") int pageLength) {
+            @ApiParam(value = "The length of each page.", required = false) @QueryParam("pagelength") int pageLength,
+            @ApiParam(value = "Gets one value before and after the requested period.", required = false) @QueryParam("boundary") boolean boundary) {
+
+        return getItemHistoryBean(serviceName, itemName, startTime, endTime, pageNumber, pageLength, boundary);
+    }
+
+    @GET
+    @Path("/items")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets a list of items available via a specific persistence service.", response = String.class, responseContainer = "List")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK"))
+    public Response httpGetPersistenceServiceItems(@Context HttpHeaders headers,
+            @ApiParam(value = "Name of the persistence service. If not provided the default service will be used", required = false) @QueryParam("servicename") String serviceName) {
+
+        return getServiceItemList(serviceName);
+    }
 
-        return getItemHistoryBean(serviceName, itemName, startTime, endTime, pageNumber, pageLength);
+    @DELETE
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Delete item data from a specific persistence service.", response = String.class, responseContainer = "List")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 400, message = "Invalid filter parameters"),
+            @ApiResponse(code = 404, message = "Unknown persistence service") })
+    public Response httpDeletePersistenceServiceItem(@Context HttpHeaders headers,
+            @ApiParam(value = "Name of the persistence service.", required = true) @QueryParam("servicename") String serviceName,
+            @ApiParam(value = "The item name.", required = true) @PathParam("itemname") String itemName,
```

Comment:
Might we also need to have a url without having to provide an item name? For functionality like "delete all data older than 12 months"? Or is this too risky to expose through REST or not a usecase for now?